### PR TITLE
handle revocation in getClient

### PR DESCRIPTION
**Pull request by enguerran**
The revocation is only handle in the oauthFlow, I think it is to late to catch the error.
Maybe we can remove the fix by @nono on line 287.

### CODE REVIEW

**Comment by enguerran** [comment]
It is mandatory for https://github.com/cozy/cozy-files-v3/pull/138

File: src/auth_v3.js [hunk 1]
```diff
@@ -149,12 +149,25 @@ export function getClient (cozy, clientParams) {
...
+    })
+    .catch((err) => {
+      // If we fall into an error while fetching the client (because of a
+      // bad connectivity for instance), we do not bail the whole process
+      // since the client should be able to continue with the persisted
+      // client and token.
+      //
+      // If it is an explicit Unauthorized error though, we bail, clear th
+      // cache and retry.
+      if (FetchError.isUnauthorized(err) || FetchError.isNotFound(err)) {
```
Comment (nono) [inline src/auth_v3.js]: `cozyFetchJSON` is used for many things and a 404 error does not mean that the client has been revoked in most cases

Comment (enguerran) [inline src/auth_v3.js]: Ok, i'll find another way.

Comment (enguerran) [inline src/auth_v3.js]: @nono:
I put the `catch` code in `getClient` https://github.com/cozy/cozy-client-js/blob/master/src/auth_v3.js#L147-L158 not in `cozyFetchJSON`.
Do you really think there is many things and a 404 error in `getClient` could be something else than a revocation?

Comment (nono) [inline src/auth_v3.js]: If `getClient` returns a 404, it means that the client does not exist. Either it has been revoked, or the client has never existed on this instance (if you take the client_id of one instance and call `getClient` on another instance for example). The effect is similar: you can't use the client and you should treat that as a revoked client.

**Comment by enguerran** [comment]
Just realized I erase the line https://github.com/cozy/cozy-client-js/blob/master/src/auth_v3.js#L157...

Now I fixed it and the build is happier.

😜

**Pull request merged**